app/src/androidTest/java/net/aiscope/gdd_app/ui/metadata/SampleImagesAdapterTest.kt

```diff
@@ -0,0 +1,69 @@
+package net.aiscope.gdd_app.ui.metadata
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import kotlinx.coroutines.runBlocking
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import java.io.InputStream
+
+@RunWith(AndroidJUnit4::class)
+class SampleImagesAdapterTest {
+    private val testContext = InstrumentationRegistry.getInstrumentation().context
+    private val targetContext = InstrumentationRegistry.getInstrumentation().targetContext
+    private val cacheDir = targetContext.cacheDir
+    private lateinit var targetFile: File
+
+    @Before
+    fun setup() {
+        targetFile = File(targetContext.filesDir, "test.bmp");
+        targetFile.createNewFile()
+
+        val stream = testContext.assets.open("w3c_home.bmp");

```

Comment:
In `MaskActivityTest.kt` there's a `getAssetStream` method that does this, and even has a failback. I think that it could be made an extension function on `Context`, and be reused here and in `BitmapReaderTest` as well, what do you think?

Reply:
Done. Also moved the `ParcelableExt` to a new `test.extensions` package